test/throttler/remote_throttler.js

```diff
@@ -0,0 +1,171 @@
+  let serviceName = 'service';
+  let operation = 'op';
+  let other_operation = 'oop';
+  let uuid = 'uuid';
+
+  before(() => {
+    server = new ConfigServer().start();
+  });
+
+  after(() => {
+    server.close();
+  });
+
+  beforeEach(() => {
+    server.clearConfigs();
+    logger = new MockLogger();
+    metrics = new Metrics(new LocalMetricFactory());
+    throttler = new RemoteThrottler(serviceName, {
+      refreshIntervalMs: 0,
+      initialDelayMs: 60000,
+      metrics: metrics,
+      logger: logger,
+    });
+  });
+
+  afterEach(() => {
+    throttler.close();
+  });
+
+  it('return false for isAllowed on initial call and return true once credits are initialized', done => {
+    throttler.setProcess({ uuid: uuid });
+    server.addCredits(serviceName, [{ operation: operation, credits: 3 }]);
+    throttler._onCreditsUpdate = _throttler => {
+      assert.isOk(_throttler.isAllowed(operation));
+      assert.equal(_throttler._credits[operation], 2);
+      assert.equal(LocalBackend.counterValue(metrics.throttlerUpdateSuccess), 1);
+      done();
+    };
+    assert.isNotOk(throttler.isAllowed(operation));
+    throttler._refreshCredits();
+  });
+
+  it('log an error if _refreshCredits is called prior to UUID being set', () => {
+    throttler._refreshCredits();
+    assert.equal(logger._errorMsgs.length, 1);
+  });
+
+  it('not fetch credits if uuid is invalid', () => {
+    throttler.setProcess({ uuid: null });
+    throttler._refreshCredits();
+    assert.equal(logger._errorMsgs.length, 1, `errors=${logger._errorMsgs}`);
+  });
+
+  it("return false for _isAllowed if operation isn't in _credits or operation has no credits", () => {
+    assert.isNotOk(
+      throttler._isAllowed(operation),
+      'operation is not set so operation should not be allowed'
+    );
+    throttler._credits[operation] = 0;
+    assert.isNotOk(throttler._isAllowed(operation), 'operation is set but lacks credit');
+  });
+
+  it("return false for isAllowed if operation doesn't have enough credits", () => {
+    throttler._credits[operation] = 0.5;
+    assert.isNotOk(throttler._isAllowed(operation));
+  });
+
+  it('succeed when we retrieve credits for multiple operations', done => {
+    throttler.setProcess({ uuid: uuid });
+    server.addCredits(serviceName, [
+      { operation: operation, credits: 5 },
+      { operation: other_operation, credits: 3 },
+    ]);
+    throttler._credits[operation] = 0;
+    throttler._credits[other_operation] = 0;
+    throttler._onCreditsUpdate = _throttler => {
+      assert.isOk(_throttler.isAllowed(operation));
+      assert.equal(_throttler._credits[operation], 4);
+      assert.isOk(_throttler.isAllowed(other_operation));
+      assert.equal(_throttler._credits[other_operation], 2);
+      assert.equal(LocalBackend.counterValue(metrics.throttlerUpdateSuccess), 1);
+      done();
+    };
+    throttler._refreshCredits();
+  });
+
+  it('emit failure metric on failing to query for credits', done => {
```

Comment:
For the failure scenarios, you might consider also ensuring the credit update callback does not fire.

Reply:
stupid setTimeout issue on travis strikes again. I think asserting that the logs were added are good enough for these tests.